src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewDataDictionary.cs

```diff
@@ -41,9 +42,15 @@ public class ViewDataDictionary : IDictionary<string, object>
         /// </summary>
         /// <param name="source"><see cref="ViewDataDictionary"/> instance to copy initial values from.</param>
         /// <remarks>
+        /// <para>
         /// For use when copying a <see cref="ViewDataDictionary"/> instance and the declared <see cref="Model"/>
         /// <see cref="Type"/> will not change e.g. when copying from a <see cref="ViewDataDictionary{TModel}"/>
         /// instance to a base <see cref="ViewDataDictionary"/> instance.
+        /// </para>
+        /// <para>
+        /// This constructor should not be used in any context where <see cref="Model"/> may be set to a value
+        /// incompatible with the declared type of <paramref name="source"/>.
+        /// </para>
         /// </remarks>
         public ViewDataDictionary(ViewDataDictionary source)
```

Comment:
@Eilon this constructor is unsafe because it preserves the source's `_declaredModelType`. It's also no longer used in our product code. Any objection to removing it completely?